### ISSUE TITLE
Co-exist with metamask

### ISSUE THREAD

**Issue by naps62** [iron-wallet/iron#70]
This issue is mostly for brainstorming and discussion, but the main goal is to remove the need to run the extension in a separate profile without metamask.

## Context

So far I've been using a separate chrome profile to run the extension.
Since both Metamask and this inject `window.ethereum`, I wanted to avoid conflicts there.

Truth be told, I never actually tried to see what happens if both run on the same profile. I'm guessing it'll be dependant on whichever extension ends up executing first? And I don't know if that's even deterministic or not

However, it should be theoretically possible to have them coexist in some way. A naive approach would be to defer our own injection by a few miliseconds, to ensure we always override Metamask

Then we can think more inteligently: maybe we only want to override for certain domains such as localhost? or maybe we want to give users that choice through the extension configs?

**Comment by pvgo80** [comment]
> Truth be told, I never actually tried to see what happens if both run on the same profile. I'm guessing it'll be dependant on whichever extension ends up executing first? And I don't know if that's even deterministic or not

After some testing (the documentation was not entirely clear), it seems main factor is the order of installation. Whichever application is installed first executes first. I installed Iron app first and then Metamask, so I had to disable Iron in order to use Metamask. 

ChatGPT and a random post on reddit suggest that "specific APIs or the use of declarative content scripts" could lead to changes on priority.
